packages/admin/src/pages/super-admin-dashboard/user/[id]/change-roles.test.tsx

```diff
@@ -68,4 +86,82 @@ describe('Edit role page', () => {
       screen.getByText('Applicant').parentElement?.previousSibling
     ).not.toBeChecked();
   });
+
+  test('Should redirect to change department page if new Admin user', async () => {
+    mockParseBody.mockResolvedValue({ newUserRoles: ['3', '4'] });
+    mockGetUserById.mockReturnValue({
+      firstName: 'john',
+      lastName: 'm',
+      organisationName: 'tco',
+      emailAddress: 'superAdmin@and.digital',
+      roles: [getMockRoles()[0], getMockRoles()[1]],
+      department: null,
+      created: 'NULL',
+    });
+
+    const getDefaultContext = () => ({
+      params: { id: '1' },
+      req: { method: 'POST' },
+    });
+    const result = await getServerSideProps(getContext(getDefaultContext));
+    expect(result).toEqual({
+      redirect: {
+        destination: '/super-admin-dashboard/user/1/change-department',
+        statusCode: 302,
+      },
+    });
+  });
+
+  test('Should redirect to account page as User who is an Applicant', async () => {
+    mockParseBody.mockResolvedValue({ newUserRoles: ['1', '2'] });
+    mockGetUserById.mockResolvedValue({
+      firstName: 'john',
+      lastName: 'm',
+      organisationName: 'tco',
+      emailAddress: 'superAdmin@and.digital',
+      roles: [getMockRoles()[0], getMockRoles()[1]],
+      department: null,
+      created: 'NULL',
+    });
+    const getDefaultContext = () => ({
+      params: { id: '1' },
+      req: { method: 'POST' },
+    });
+    const result = await getServerSideProps(getContext(getDefaultContext));
+    expect(result).toEqual({
+      redirect: {
+        destination: '/super-admin-dashboard/user/1',
+        statusCode: 302,
+      },
+    });
+  });
+
+  test('Should redirect to account page as User who is already Admin (with a Department)', async () => {
+    mockParseBody.mockResolvedValue({ newUserRoles: ['1', '2', '3'] });
+    mockGetUserById.mockResolvedValue({
+      firstName: 'john',
+      lastName: 'm',
+      organisationName: 'tco',
+      emailAddress: 'superAdmin@and.digital',
+      roles: [
+        getMockRoles()[0],
+        getMockRoles()[1],
+        getMockRoles()[2],
+        getMockRoles()[3],
+      ],
```

Comment:
this is equivalent to

```suggestion
      roles: getMockRoles(),
```